### PR TITLE
[Windows] Check PYTHON_INSTALL_DIR before making directory.

### PR DESCRIPTION
**Pull request by seanyen**
This change is to reduce the `stderr` outputs from a common code path. For building packages in a merged space, the code path to unconditionally create `Lib\site-packages` (by `PYTHON_INSTALL_DIR`) will result in complaints  `A subdirectory or file src already exists.` to `stderr` for the second and later packages exercising this flow.

To reduce the noise to developers, this pull request is to only make directory when it doesn't exist.

Thanks for review and please consider to back-port to `kinetic-devel` if it makes sense.

### CODE REVIEW

File: cmake/templates/python_distutils_install.bat.in [hunk 1]
```diff
@@ -7,7 +7,9 @@ if DEFINED DESTDIR (
 
 cd "@INSTALL_CMD_WORKING_DIRECTORY@"
 
-mkdir "@SETUPTOOLS_INSTALL_PREFIX@\@PYTHON_INSTALL_DIR@"
+if NOT EXIST "@SETUPTOOLS_INSTALL_PREFIX@\@PYTHON_INSTALL_DIR@\" (
```
Comment (dirk-thomas) [inline cmake/templates/python_distutils_install.bat.in:10]: The backslash only on the closing quote but not on the opening one looks weird. Are you sure about this?

Comment (seanyen) [reply cmake/templates/python_distutils_install.bat.in:10]: The backslash wasn't for character escaping and it is part of the string. Ending with the path separator will make the batch script to check the directory existence, and only if the directory does exist, it skips the directory creation.

**Comment by dirk-thomas** [comment]
Thanks for the patch.

**Comment by dirk-thomas** [comment]
Cherry-picked to `kinetic-devel` in cc395971e65658e7dd0d45b8f1e7f4eff706e016.

**Pull request merged**